src/NetStreams/Internal/NetStream.cs

```diff
@@ -81,13 +75,22 @@ public Task StartAsync(CancellationToken token)
                         _onError(ex);
                         if (!_configuration.ContinueOnError)
                         {
-                            Stop();
+                            ResetOffset(consumeResult);
                         }
                     }
                 }
             }, token, TaskCreationOptions.LongRunning, TaskScheduler.Default).Unwrap();
         }
 
+        private void ResetOffset(ConsumeResult<TKey, TMessage> consumeResult)
+        {
+            if (consumeResult != null)
+            {
+                _log.Debug($"Resetting offset to topic: {consumeResult.TopicPartitionOffset.Topic}, partition:{consumeResult.TopicPartition.Partition}, offset: {consumeResult.TopicPartitionOffset.Offset}");
+                _consumer.Seek(consumeResult.TopicPartitionOffset);
```

Comment:
Not sure how this will work in eviCore's new cloud-hosted Kafka clusters.  The docker-compose file doesn't have ACLs turned-on.  While on the Top Gun team we learned that ACLs are being required not just for the topics, but also for the consumer groups.  (Your service authenticates as a "user principal", which must be given access to both the topics and the consumer group it will use.)

Have you tried this on a cluster that requires ACLs?

Reply:
Could you just not commit the message, like eviCore.Bus does?

Reply:
Good idea to try it, but conceptually it seems like Seek should be the same as Read+Commit. I will be sad panda if you can Read and Commit but not Seek on the same topic.